File: 80_Remove_Duplicates_from_Sorted_Array-II.java

// Time Complexity : O(length of array)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode :
// Three line explanation of solution in plain english

// Your code here along with comments explaining your approach

class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length<2)
            return nums.length;
        int index=2,i=2,flg=nums[1]==nums[0]?1:0,prev=nums[1];
        while(i<nums.length){
            if(nums[i]==nums[i-1]){
                if(flg==0){
                    flg=1;
                    nums[index++]=nums[i];
                }
            }else{
                nums[index++]=nums[i];
                flg=0;
            }
            i++;
        }
        
        return index;
    }
}

File: 80_Remove_Duplicates_from_Sorted_Array_II.py
# Time Complexity : O(n) [n = numner of elements in list]
# 
# Space Complexity : O(1)
# 
# Did this code successfully run on Leetcode : Yes
# 
# Any problem you faced while coding this : No
# 
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        write = 0
        index = 1
        start, end = 0, 0
        while end < len(nums):
            if nums[start] == nums[end]:
                if end - start < 2: # The write is incremented and updated at most twice for the same character
                    nums[write] =  nums[start]
                    write+=1
                end+=1
            else:
                start=end
        return write